DatosAssignment2_[Ruben]

rubzen opened the pull request.

rubzen committed:
Add files via upload

rovalcor commented on DatosAssignment2_[Ruben]:
```diff
@@ -0,0 +1 @@
+<mxfile userAgent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36" version="8.5.7" editor="www.draw.io" type="google"><diagram id="fe6295e1-77d3-4cf7-30d9-1b711ddc5765" name="Page-1">7V3bkps4EP0aP06KO/Zj7J1Jajepncpka7NPKRlkmw1GXpAznnz9SiBxkcAmDiBmhiRVMY3Qpftw1N0IMTNX+9O7GBx2H5EPw5mh+aeZ+dvMMHRtMSf/UclTJnE1OxNs48BnhQrBQ/AD8 ...
```
Creo que esto se pegó por accidente :(

rovalcor commented on DatosAssignment2_[Ruben].sql:
```diff
@@ -0,0 +1,23 @@
+CREATE TABLE IF NOT EXISTS student(
+	id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
```
La indentación está dispareja con los PRIMARY KEYs de las tablas, no van a la par con las otras columnas.

rovalcor commented on DatosAssignment2_[Ruben].sql:
```diff
@@ -0,0 +1,23 @@
+CREATE TABLE IF NOT EXISTS student(
```
Usualmente el nombre de la tabla va en plural. Eso también aplica para todas las tablas.

rovalcor commented on DatosAssignment2_[Ruben].sql:
```diff
@@ -0,0 +1,23 @@
+	id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(30) NOT NULL,
+    curp VARCHAR(18) NOT NULL UNIQUE,
+    count_number VARCHAR(10) NOT NULL UNIQUE
```
Solo una observación muy pequeña: creo que aquí más bien se escribe `account_number`

rovalcor commented on DatosAssignment2_[Ruben].sql:
```diff
@@ -0,0 +1,23 @@
+    name VARCHAR(30) NOT NULL,
+    topic VARCHAR(30),
+    courses_id INT NOt NULL,
+    FOREIGN KEY (courses_id) REFERENCES couse(id)
```
Hay un typo, dice `couse(id)` cuando debe hacer referencia a `course`. No te falló?

rovalcor commented on DatosAssignment2_[Ruben].sql:
```diff
@@ -0,0 +1,23 @@
+CREATE TABLE IF NOT EXISTS student_course(
+	students_id VARCHAR(30),
+    courses_id VARCHAR(30),
+    date DATE NOT NULL
```
`date` es un nombre muy genérico para una columna. ¿Qué significa este date? ¿Fecha de creación o de actualización?

rovalcor commented on DatosAssignment2_[Ruben].sql:
```diff
@@ -0,0 +1,23 @@
+    FOREIGN KEY (courses_id) REFERENCES couse(id)
+);
+CREATE TABLE IF NOT EXISTS student_course(
+	students_id VARCHAR(30),
```
Usualmente yo le pondría un constraint de NOT NULL a las columnas de las tablas intermedias, porque un registro sin estudiante o sin curso no tiene mucho sentido. También suelo ponerle el nombre de la tabla en singular para este caso.

rovalcor commented on DatosAssignment2_[Ruben].sql:
```diff
@@ -0,0 +1,23 @@
+    courses_id INT NOt NULL,
+    FOREIGN KEY (courses_id) REFERENCES couse(id)
+);
+CREATE TABLE IF NOT EXISTS student_course(
```
Siguiendo lo de la pluralidad, diría que sería mejor `students_courses`

rovalcor commented on DatosAssignment2_[Ruben].sql:
```diff
@@ -0,0 +1,23 @@
+	id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(30) NOT NULL,
+    topic VARCHAR(30),
+    courses_id INT NOt NULL,
```
Iría por el nombre en singular también.

rovalcor reviewed (changes requested):
Los comentarios que te dejo más que en el diagrama relacional falta cardinalidad del lado de `students.id` y `courses.id`

rubzen committed:
Versión Actualizada

Hola, abajo del código original agregué las modificaciones.
Gracias por tus comentarios.

rovalcor commented on DatosAssignment2_[Ruben].sql:
```diff
@@ -0,0 +1,49 @@
+	student_id VARCHAR(30) NOT NULL,
+course_id VARCHAR(30) NOT NULL,
+datecourse DATE NOT NULL
+);
```
Me vas a querer matar: olvidé comentarte la vez anterior que faltan los foreign keys de esta tabla. Solo falta eso y ya puedo darte la tarea como válida.

rovalcor reviewed (changes requested):
Solo faltan unos foreign keys y estarás listo

rubzen committed:
Add files via upload

Hola Rudy, he realizado unos cambios de acuerdo a tus comentarios. 
Quedo al pendiente.
Saludos y muchas gracias !!

rubzen committed:
Update DatosAssignment2_[Ruben].sql

Hola, gracias por confirmar los cambios. Saludos !!!

rovalcor reviewed (approved):
🎉
